ddcommon/src/container_id.rs

```diff
@@ -85,10 +89,27 @@ fn extract_container_id(filepath: &Path) -> Result<String, Box<dyn std::error::E
     Err(ContainerIdNotFoundError.into())
 }
 
+/// # Safety
+/// Must not be called in multi-threaded contexts
+pub unsafe fn set_cgroup_file(file: String) {
+    TESTING_CGROUP_PATH = Some(file)
+}
+
+fn get_cgroup_path() -> PathBuf {
+    // Safety: we assume set_cgroup_file is not called when it shouldn't
+    if let Some(path) = unsafe { TESTING_CGROUP_PATH.as_ref() } {
```

Comment:
@bwoebi would making this an env variable work as well ?

Reply:
An env var would be okay as well I think.

Reply:
To be honest, I'm not a big fan of the env variable approach. The current approach adds a new API, which the application needs to opt in to for using (e.g. when it's running the tests), and I like that a bit more than using an environment variable where suddenly we've created a public API that any user of libdatadog will "inherit" and make available in product because the environment variable can be set by customers.

So... I dislike coming into reviews and giving advice that goes in the exact other direction, but @pawelchcki can I convince you that an env variable may not be the best approach here?

Reply:
Those were some good points - @ivoanjo consider me convinced.